### ISSUE TITLE
App list is empty

### ISSUE THREAD

**Issue by mizzunet** [MuntashirAkon/AppManager#1196]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [ ] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

No  apps are showing in the app list. No filter has benn selected

### To Reproduce

No specific steps, simply opening the app

### Expected behavior

Apps are be shown

### Screenshots

![Screenshot_20230921-183648](https://github.com/MuntashirAkon/AppManager/assets/10193999/ab24fea9-c49b-41b6-8d40-78dc72d20435)

### Logs

<details>
<summary>logs</summary>

```
21.09 18:37:34.701 10228 21833 21833 io.github.muntashirakon.AppManager E/akon.AppManager: Not starting debugger since process cannot load the jdwp agent.
21.09 18:37:34.729 10228 21833 21833 io.github.muntashirakon.AppManager D/CompatibilityChangeReporter: Compat change id reported: 171979766; UID 10228; state: ENABLED
21.09 18:37:34.746 10228 21833 21833 io.github.muntashirakon.AppManager V/GraphicsEnvironment: ANGLE Developer option for 'io.github.muntashirakon.AppManager' set to: 'default'
21.09 18:37:34.747 10228 21833 21833 io.github.muntashirakon.AppManager V/GraphicsEnvironment: ANGLE GameManagerService for io.github.muntashirakon.AppManager: false
21.09 18:37:34.747 10228 21833 21833 io.github.muntashirakon.AppManager V/GraphicsEnvironment: Neither updatable production driver nor prerelease driver is supported.
21.09 18:37:34.748 10228 21833 21833 io.github.muntashirakon.AppManager D/NetworkSecurityConfig: No Network Security Config specified, using platform default
21.09 18:37:34.748 10228 21833 21833 io.github.muntashirakon.AppManager D/NetworkSecurityConfig: No Network Security Config specified, using platform default
21.09 18:37:34.773 10228 21833 21833 io.github.muntashirakon.AppManager D/AppCompatDelegate: Checking for metadata for AppLocalesMetadataHolderService : Service not found
21.09 18:37:34.776 10228 21833 21833 io.github.muntashirakon.AppManager E/akon.AppManager: Invalid ID 0x00000000.
21.09 18:37:34.786 10228 21833 21833 io.github.muntashirakon.AppManager D/CompatibilityChangeReporter: Compat change id reported: 210923482; UID 10228; state: ENABLED
21.09 18:37:34.812 10228 21833 21833 io.github.muntashirakon.AppManager D/CompatibilityChangeReporter: Compat change id reported: 237531167; UID 10228; state: DISABLED
21.09 18:37:34.816 10228 21833 21854 io.github.muntashirakon.AppManager W/Parcel: Expecting binder but got null!
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: QUALCOMM build                   : 85da404, I46ff5fc46f
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: Build Date                       : 11/30/20
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.31.04.01
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: Local Branch                     : promo490_3_Google
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: Remote Branch                    : 
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: Remote Branch                    : 
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: Reconstruct Branch               : 
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: Build Config                     : S P 10.0.4 AArch64
21.09 18:37:34.833 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
21.09 18:37:34.839 10228 21833 21854 io.github.muntashirakon.AppManager I/AdrenoGLES-0: PFP: 0x016ee190, ME: 0x00000000
21.09 18:37:34.841 10228 21833 21854 io.github.muntashirakon.AppManager W/AdrenoUtils: <ReadGpuID_from_sysfs:197>: Failed to open /sys/class/kgsl/kgsl-3d0/gpu_model
21.09 18:37:34.841 10228 21833 21854 io.github.muntashirakon.AppManager W/AdrenoUtils: <ReadGpuID:221>: Failed to read chip ID from gpu_model. Fallback to use the GSL path
21.09 18:37:34.847 10228 21833 21854 io.github.muntashirakon.AppManager E/OpenGLRenderer: Unable to match the desired swap behavior.
21.09 18:37:34.850 10228 21833 21860 io.github.muntashirakon.AppManager I/Gralloc4: Adding additional valid usage bits: 0x0
21.09 18:37:34.922 10228 21833 21833 io.github.muntashirakon.AppManager D/CompatibilityChangeReporter: Compat change id reported: 171228096; UID 10228; state: ENABLED
21.09 18:37:34.927 10228 21833 21854 io.github.muntashirakon.AppManager W/Parcel: Expecting binder but got null!
21.09 18:37:34.937 10228 21833 21854 io.github.muntashirakon.AppManager W/Parcel: Expecting binder but got null!
21.09 18:37:34.950 10228 21833 21854 io.github.muntashirakon.AppManager E/OpenGLRenderer: Unable to match the desired swap behavior.
21.09 18:37:34.984 10228 21833 21854 io.github.muntashirakon.AppManager E/OpenGLRenderer: Unable to match the desired swap behavior.
21.09 18:37:34.999 10228 21833 21833 io.github.muntashirakon.AppManager D/CompatibilityChangeReporter: Compat change id reported: 163400105; UID 10228; state: ENABLED
21.09 18:37:35.001 10228 21833 21833 io.github.muntashirakon.AppManager I/AssistStructure: Flattened final assist data: 3440 bytes, containing 2 windows, 25 views
21.09 18:37:35.732 10228 21833 21854 io.github.muntashirakon.AppManager D/OpenGLRenderer: endAllActiveAnimators on 0xb4000070d8ebdcb0 (RippleDrawable) with handle 0xb400006fe8ee6cb0
21.09 18:37:35.799 10228 21833 21833 io.github.muntashirakon.AppManager D/CompatibilityChangeReporter: Compat change id reported: 78294732; UID 10228; state: ENABLED
21.09 18:37:35.893 10228 21833 21876 io.github.muntashirakon.AppManager D/RootServiceManager: ( [ -d  /data/local/tmp ] || ( rm /data/local/tmp; mkdir /data/local/tmp && chmod 771 /data/local/tmp && chown 2000:2000 /data/local/tmp ) ) && cp /storage/emulated/0/Android/data/io.github.muntashirakon.AppManager/cache/main.jar /data/local/tmp && chmod 755 /data/local/tmp/main.jar && chown shell:shell /data/local/tmp/main.jar && (CLASSPATH=/data/local/tmp/main.jar  /proc/21833/exe  /system/bin --nice-name=io.github.muntashirakon.AppManager:root io.github.muntashirakon.AppManager.server.RootServiceMain io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.ipc.AMService 10228 3e0cb32f-b972-4d54-a919-3406c7e37a2b start)&
21.09 18:37:36.264 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.266 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.268 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.270 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.272 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.274 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.275 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.277 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.279 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.280 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.282 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.283 10228 21833 21833 io.github.muntashirakon.AppManager I/MaterialCardView: Setting a custom background is not supported.
21.09 18:37:36.303 10228 21833 21854 io.github.muntashirakon.AppManager W/Parcel: Expecting binder but got null!
21.09 18:37:36.317 10228 21833 21854 io.github.muntashirakon.AppManager W/Parcel: Expecting binder but got null!
21.09 18:37:36.329 10228 21833 21854 io.github.muntashirakon.AppManager E/OpenGLRenderer: Unable to match the desired swap behavior.
21.09 18:37:36.342 10228 21833 21854 io.github.muntashirakon.AppManager E/OpenGLRenderer: Unable to match the desired swap behavior.
21.09 18:37:38.764 10228 21833 21833 io.github.muntashirakon.AppManager D/CompatibilityChangeReporter: Compat change id reported: 160794467; UID 10228; state: ENABLED
21.09 18:37:38.777 10228 21833 21854 io.github.muntashirakon.AppManager D/OpenGLRenderer: endAllActiveAnimators on 0xb4000070d8f0f110 (RippleDrawable) with handle 0xb400006fe8ed0240
```
</details>

### Device info

- Device: Pixel 4a
- OS Version: CalyxOS Android 13
- App Manager Version: Version 3.1.1
- Mode: Root

**Comment by MuntashirAkon** [comment]
Try without root first and send me the logs, preferably, in a file.

**Comment by MuntashirAkon** [comment]
Your logs are incomplete. No logs before 12:28:53, but your screenshot says the time was 12:27:xx when App Manager reported an error.

**Comment by MuntashirAkon** [comment]
> https://oshi.at/SLyu

404 Not found.

**Comment by Stu75e** [comment]
I also have this problem in version v4.0.0-alpha02

**Comment by EdRoxter** [comment]
Just for the record, after upgrading from 3.1.1 to 3.1.2 (F-Droid), I see the same issue on my Samsung S10e, Android 12/OneUI4.1 with Magisk 26.3. Downgrading to 3.1.1 makes it work again.

Can I provide anything to narrow down the issue?

**Comment by MuntashirAkon** [comment]
> Can I provide anything to narrow down the issue?

1. Install a debug version from here: https://github.com/MuntashirAkon/AppManager/actions/runs/6233985576 (for v4.0.0-alphaxx users, install the latest debug instead)
2. Start taking logs using a third-party log taker (you can also use the other App Manager)
3. Start the app
4. Wait until it finishes initialising and then wait another 20 seconds for background loading of apps
5. Stop taking logs
6. Upload the logs here (or via E-Mail if you think the logs contain anything that may affect your privacy).

**Comment by MuntashirAkon** [comment]
> This is not downloadable. Says "Expired"

[AppManager_v3.1.2-DEBUG#2589.zip](https://github.com/MuntashirAkon/AppManager/files/12772641/AppManager_v3.1.2-DEBUG.2589.zip)

**Comment by shuvashish76** [comment]
Fresh install AM v3.1.2, device details provided, mode : no root.
For me App list is empty & running apps loading forever (similar to #1138)
[am_file.zip](https://github.com/MuntashirAkon/AppManager/files/12773413/am_file.zip)
Logs recorded > refreshed main page multiple times > tried troubleshooting > checked running apps list
Let me know if it has any useful infos

Note: No issues on v3.1.1 but both issue reappears when updating to 3.1.2

**Comment by romantisch1989** [comment]
fixes for this problem so that the system applications are seen by the phone all
adding what is not displayed is probably free.Adb debugging , everything was included in App Manager 3.1.1 and in App Manager version 4.0.0 alpha 01

**Comment by MuntashirAkon** [comment]
> running apps loading forever

Thanks for reporting this particular issue. I was able to reproduce this particular issue. A fix will be available in the Debug channel soon for you to test.

> Let me know if it has any useful infos

Your logs seem to be partial. You cannot take the logs from AM Debug itself because it may miss initial logs which are important in order to figure out the original issue.

**Comment by MuntashirAkon** [comment]
> fixes for this problem so that the system applications are seen by the phone all
> adding what is not displayed is probably free.Adb debugging , everything was included in App Manager 3.1.1 and in App Manager version 4.0.0 alpha 01

Fix is possible only if the cause is known. Since I cannot reproduce it, I cannot fix it without some help from those who're affected. From what I can see, the issue might be related to some `SecurityException` issue which may involve fetching packages for an user that cannot be accessed by App Manager in no-root mode.

**Comment by shuvashish76** [comment]
> You cannot take the logs from AM Debug itself because it may miss initial logs which are important in order to figure out the original issue.

Logs taken from AM v3.1.2 stable release itself. But ok got it, next time I'll record noroot mode dubug build with 3rd party apps such as logfox.

**Comment by shuvashish76** [comment]
Here : 
[30-09_17-15-51.txt](https://github.com/MuntashirAkon/AppManager/files/12774567/30-09_17-15-51.txt)

- Some apps shown in the list then after refreshing it disappeared. This happened couple of times.
- App Usage is empty for Today and Weekly but works for Yesterday.
- Only 2 running apps in the list sh and sh.

**Comment by MuntashirAkon** [comment]
> Here :
> [30-09_17-15-51.txt](https://github.com/MuntashirAkon/AppManager/files/12774567/30-09_17-15-51.txt)

```
java.lang.IllegalStateException: Retrieved 348 packages out of 347 applications which is impossible
	at io.github.muntashirakon.AppManager.compat.PackageManagerCompat.getInstalledPackages(PackageManagerCompat.java:105)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:256)
	at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:148)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.getInstalledOrBackedUpApplicationsFromDb(PackageUtils.java:170)
	at io.github.muntashirakon.AppManager.main.MainViewModel.lambda$loadApplicationItems$3$io-github-muntashirakon-AppManager-main-MainViewModel(MainViewModel.java:307)
...
```

This seems to be causing the issue in v3.1.2 (unrelated to the original issue reported by @mizzunet). Not sure who the culprit is, but it could be a static library such as trichrome.

**Comment by MuntashirAkon** [comment]
@shuvashish76: Please check the latest debug.

**Comment by shuvashish76** [comment]
Same issues, nothing changed for me.
AM Debug v3.1.2 Run#2598
[30-09_22-10-35.txt](https://github.com/MuntashirAkon/AppManager/files/12775148/30-09_22-10-35.txt)

**Comment by MuntashirAkon** [comment]
> Same issues, nothing changed for me.

It's okay. This was a test build.

> [30-09_22-10-35.txt](https://github.com/MuntashirAkon/AppManager/files/12775148/30-09_22-10-35.txt)

```
Loaded extra packages: [com.samsung.upsmtheme]
```

What's this mystery package `com.samsung.upsmtheme`?

**Comment by shuvashish76** [comment]
> What's this mystery package `com.samsung.upsmtheme`?

ADL [description](https://github.com/MuntashirAkon/android-debloat-list/blob/0d5369a128f0ad5fdbebf0da6b466d551f43c5c5/oem.json#L5839) :  "Handle the theme of UPSM (Ultra Power Saving Mode). Safe to remove."

Many samsung packages were disabled by me including this one. Seems can't unfreeze it from AM, which is different offtopic issue. I don't think a theme app has any relevance here as AM v3.1.1 works without any issue.

**Comment by MuntashirAkon** [comment]
> Many samsung packages were disabled by me including this one. Seems can't unfreeze it from AM, which is different offtopic issue. I don't think a theme app has any relevance here as AM v3.1.1 works without any issue.

This particular package is causing the issue in your case (every log item related to missing packages mentioned this package for no reason). That's why asking.

The issue should now be fixed by 4c785be365ac58e56b4fc925d920e657b3d5abcd, but the underlying issue is still unknown. However, this should not fix the original issue by @mizzunet which should be entirely different (but no guarantees since the underlying issue is yet to be identified).

**Comment by shuvashish76** [comment]
Thanks. [issues](https://github.com/MuntashirAkon/AppManager/issues/1196#issuecomment-1741750015)[1, 3 - ✔️, 2 - now all empty ❌ needs a separate issue]
Checked with both no root & adb mode, uninstall-reinstalled few times app list works as expected for me.👍🏿

**Comment by MuntashirAkon** [comment]
> 2 - now all empty ❌

Can you share some logs taken from that page?

**Comment by shuvashish76** [comment]
AppManager_v3.1.2-DEBUG#2600
<details><summary>Steps</summary>
<p>

1. Fresh install the debug version > Open
2. Open log fox > Add filter > pick AM > Save 
3. Open AM > Click on 3 dot menu
4. Logfox > Clear logs if present > Start logs record (_so that it take logs for AM App Usage only_)
5. Now open AM from device recents > click on **App Usage** > test whatever needs to be checked
6. Logfox > Stop recording & share logs

</p>
</details> 

[01-10_08-36-09.txt](https://github.com/MuntashirAkon/AppManager/files/12776038/01-10_08-36-09.txt)

**Comment by MuntashirAkon** [comment]
> AppManager_v3.1.2-DEBUG#2600
> [01-10_08-36-09.txt](https://github.com/MuntashirAkon/AppManager/files/12776038/01-10_08-36-09.txt)

Fixed via cb72a998d5248558b7216949b33b295fc3d3958d

**Comment by MuntashirAkon** [comment]
@mizzunet: Please download the latest debug build to test and take logs. Link: https://github.com/MuntashirAkon/AppManager/actions/runs/6368164800

**Comment by directentis1** [comment]
This has been fixed on latest release of the app (3.1.3) on my Android 10, Magisk root-based.
Thanks much for your hard works!!
